Multitouch Support/Native/VoodooI2CNativeEngine.cpp

```diff
@@ -114,16 +88,57 @@ bool VoodooI2CNativeEngine::start(IOService* provider) {
     setProperty(kIOFBTransformKey, 0ull, 32);
     setProperty("VoodooInputSupported", kOSBooleanTrue);
 
-    isForceClickEnabled = true;
-    isForceTouchEmulationEnabled = true;
-    OSBoolean* enableForceTouchEmulationProperty = OSDynamicCast(OSBoolean, this->getProperty("EnableForceTouchEmulation"));
-    if (enableForceTouchEmulationProperty) {
-        isForceTouchEmulationEnabled = enableForceTouchEmulationProperty->getValue();
+
+    OSDictionary* nameMatch = IOService::serviceMatching("AppleMultitouchDevice");
```

Comment:
This seems like it could cause undefined behaviour. If more than one multitouch device is connected to the system at the same time (say a touchscreen and a trackpad) then this would only grab the first one to connect. I think it is better to grab the multitouch device that exists as a child of the current instance of the native engine and attach to that.